release/k3s/images.go

```diff
@@ -0,0 +1,159 @@
+package k3s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/rancher/ecm-distro-tools/registry"
+)
+
+const (
+	k3sImagesFile = "k3s-images.txt"
+)
+
+type ReleaseImage struct {
+	Reference         name.Reference
+	ExpectsLinuxAmd64 bool
+	ExpectsLinuxArm64 bool
+}
+
+type Image struct {
+	ReleaseImage
+	RegistryResults map[string]registry.Image
+}
+
+func (i Image) OSSImage() registry.Image {
+	if img, ok := i.RegistryResults["oss"]; ok {
+		return img
+	}
+	return registry.Image{Exists: false, Platforms: make(map[registry.Platform]bool)}
+}
+
+type ReleaseInspector struct {
+	assets     fs.FS
+	registries map[string]registry.Inspector
+	debug      bool
+}
+
+func NewReleaseInspector(assets fs.FS, registries map[string]registry.Inspector, debug bool) *ReleaseInspector {
+	return &ReleaseInspector{
+		assets:     assets,
+		registries: registries,
+		debug:      debug,
+	}
+}
+
+func (r *ReleaseInspector) InspectRelease(ctx context.Context, version string) ([]Image, error) {
+	if !strings.Contains(version, "+k3s") {
+		return nil, errors.New("only k3s releases supported")
+	}
+
+	requiredImages, err := r.imageMap(version)
+	if err != nil {
+		return nil, err
+	}
+
+	return r.checkImages(ctx, requiredImages)
+}
+
+// imageMap reads the k3s-images.txt file and creates image map
+func (r *ReleaseInspector) imageMap(version string) (map[string]ReleaseImage, error) {
+	imageMap := make(map[string]ReleaseImage)
```

Comment:
Can we rename this to remove the type from the name? Since we're in a strongly typed language, we don't need Hungarian Notation.

E.g. `images` can suffice for the scope of this method.